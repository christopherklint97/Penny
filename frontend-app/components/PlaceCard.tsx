import Image from 'next/image'
import { useRouter } from 'next/router'
import DeleteIcon from './DeleteIcon'

interface PlaceCardProps {
  img: string
  title: string
  id: string
  next?: string
  handleDelete?: (event: any) => void
}

export default function PlaceCard({
  img,
  title,
  id,
  next = 'place',
  handleDelete,
}: PlaceCardProps) {
  const router = useRouter()
  const handleClick = () => {
    next === 'trip' ? router.push(`dashboard/trips/${id}`) : null
  }

  let cursor = ''
  next === 'trip' ? (cursor = 'cursor-pointer') : (cursor = '')

  return (
    <div
      id={id}
      onClick={handleClick}
      className={`max-w-xs mx-auto transition duration-200 ease-in-out ${cursor} transform hover:-translate-y-1 hover:scale-110 hover: shadow-md p-2 rounded-xl bg-white text-center justify-center align-middle`}
    >
      <Image
        className="object-cover p-1 mr-auto rounded-lg"
        src={img}
        alt=""
        width="500"
        height="300"
        onClick={handleClick}
      />
      <h2
        className="inline-block p-2 mx-auto text-lg font-semibold"
        onClick={handleClick}
      >
        {title}
      </h2>
      {next === 'trip' ? null : <DeleteIcon handleDelete={handleDelete} />}
    </div>
  )
}
