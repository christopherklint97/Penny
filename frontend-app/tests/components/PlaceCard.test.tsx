import React from 'react'
import { render } from '../testUtils'
import PlaceCard from '../../components/PlaceCard'

describe('PlaceCard component', () => {
  it('renders without crashing', function () {
    render(
      <PlaceCard
        img="www.example.com/img"
        title="This is a title"
        next="trip"
        id="1234"
        handleDelete={jest.fn()}
      />
    )
  })

  it('matches snapshot', () => {
    const { asFragment } = render(
      <PlaceCard
        img="www.example.com/img"
        title="This is a title"
        next="trip"
        id="1234"
        handleDelete={jest.fn()}
      />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
