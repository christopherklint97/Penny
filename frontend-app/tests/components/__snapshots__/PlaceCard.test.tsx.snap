// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="max-w-xs mx-auto transition duration-200 ease-in-out cursor-pointer transform hover:-translate-y-1 hover:scale-110 hover: shadow-md p-2 rounded-xl bg-white text-center justify-center align-middle"
    id="1234"
  >
    <div
      style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
    >
      <div
        style="box-sizing: border-box; display: block; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          role="presentation"
          src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjMwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
          style="max-width: 100%; display: block;"
        />
      </div>
      <img
        alt=""
        class="p-1 mr-auto rounded-lg object-cover"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
    </div>
    <h2
      class="text-lg mx-auto font-semibold p-2 inline-block"
    >
      This is a title
    </h2>
  </div>
</DocumentFragment>
`;
